src/lib/update-alternatives/model/alternative.rb

```diff
@@ -48,13 +48,15 @@ def self.all_names
 
     # @return [Array<Alternative>] an array with all alternatives.
     def self.all
-      all_names.map { |name| load(name) }
+      all_names.map { |name| load(name) }.compact
     end
 
     # @return [Alternative] an alternative with the given name.
     # @param name [String] The name of the alternative to be loaded.
     def self.load(name)
       raw_data = Cheetah.run("update-alternatives", "--query", name, stdout: :capture).lines
+      empty_line = raw_data.find_index("\n")
+      return nil if empty_line.nil?
```

Comment:
very very tricky code. At first, I do not like returning nil. It looks a bit bogus to get names for alternatives and then such alternative cannot be query, how it can happen? Also I will consider using EmptyAlternative null object design pattern - https://en.wikipedia.org/wiki/Null_Object_pattern which can work better here and you can query alternative to method like `empty?` which can be used to filter it out.

At second it depends on having empty line. Is it really only and reliable indicator?

At third code can be simplified to use find.

So for me ideal code can look like:

```
return EmptyAlternative.new(name) unless raw_data.include?("/n")
```